test/tests/functional/pbs_mom_walltime.py

```diff
@@ -127,19 +128,51 @@ def test_suspend_time_not_counted_in_walltime(self):
         jid1 = self.server.submit(J1)
         self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
 
+        # Accumulate wall time
+        time.sleep(10)
+
         self.server.sigjob(jobid=jid1, signal="suspend")
         self.server.expect(JOB, {'job_state': 'S'}, id=jid1)
 
-        time.sleep(15)
+        # Make sure the sched cycle is completed before reading
+        # the walltime
+        self.server.manager(MGR_CMD_SET, MGR_OBJ_SERVER,
+                            {'scheduling': 'True'}, expect=True)
+        self.server.manager(MGR_CMD_SET, MGR_OBJ_SERVER,
+                            {'scheduling': 'False'}, expect=True)
+
+        jstat = self.server.status(JOB, id=jid1,
+                                   attrib=['resources_used.walltime'])
+        walltime = BatchUtils().convert_duration(
+            jstat[0]['resources_used.walltime'])
+        self.logger.info("Walltime before sleep: %d secs" % walltime)
+        self.server.manager(MGR_CMD_SET, MGR_OBJ_SERVER,
+                            {'scheduling': 'True'})
+
+        # Sleep for the job's entire walltime secs so we can catch any
+        # walltime increment during job suspension time
+        time.sleep(30)
+
+        # Used walltime should remain the same
+        self.server.expect(JOB, {'resources_used.walltime': walltime}, op=EQ,
+                           id=jid1)
+
         self.server.sigjob(jobid=jid1, signal="resume")
         self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
-
         self.server.expect(JOB, {ATTR_state: 'F'}, id=jid1, extend='x',
-                           offset=5)
-
-        # Used walltime should be less than the sleep time after suspend
-        self.server.expect(JOB, {'resources_used.walltime': 10}, op=LE,
-                           id=jid1, extend='x')
+                           offset=20)
+
+        # Verify if the job's total walltime is within limits
+        # Adding 10s buffer since min mom poll time is 10s
+        jstat = self.server.status(JOB, id=jid1,
+                                   attrib=['resources_used.walltime'],
+                                   extend='x')
+        walltime_final = BatchUtils().convert_duration(
```

Comment:
walltime_final value could be 0 if convert_duration fails or resources_used_walltime is not set for some reason. Please add a check that walltime_final is 0 then assertFail with the proper reason.